lnwallet/reservation.go

```diff
@@ -220,6 +224,14 @@ func NewChannelReservation(capacity, localFundingAmt btcutil.Amount,
 		} else {
 			chanType |= channeldb.SingleFunderBit
 		}
+
+		// If this isn't the regular internal channel funder, then this
+		// means that we won't have the funding transaction regardless
+		// of if we're the initiator or not.
+		if _, ok := fundingAssembler.(*chanfunding.WalletAssembler); !ok {
```

Comment:
hm, could the `Assembler` instead expose whether the funding tx is available? Doing it this way adding more assemblers in the future, they will all default to `NoFundingTx`

Reply:
>  Doing it this way adding more assemblers in the future, they will all default to NoFundingTx

Don't see anything wrong with this, as they may also start to define _new_ type bits themselves. As is, only the core wallet assembler has a funding transaction.

Reply:
I mean this way the reservation logic must be aware of the actual type of assembler. This won't allow me to define a new wallet assembler in an external package that satisfies the interface and also provide the funding tx. Defining it on the interface would let me do that.

Could even define a new interface 
```go
type FundingTxProvider interface {
   FundingTx func() *wire.MsgTx
}
```
and match on that here
```go
if _, ok := fundingAssembler.(chanfunding.FundingTxProvider); !ok {
    chanType |= channeldb.NoFundingTxBit
}
```

Reply:
Gotcha, implemented something like this for the current assembler. However it can't return the funding transaction at that time as it doesn't know everything it needs to do so. Instead, I've added what's more or less just a signalling method for outside callers.